### DISCUSSION TITLE
Release plan for 0.X.Y

### DISCUSSION THREAD

**Discussion by hyf0** [web-infra-dev/rspack#2879] [Core] [answered]
The release plan of Rspack before 1.0.0 is quite unclear. We would have a meeting to discuss related questions and sync up the discussion on this thread.

Some related questions:

- How to handle breaking changes?
- What's the releasing cycle?
- etc.

@web-infra-dev/rspack cc

**Comment by Boshen** [comment]
We announce 1.0.0 when we:
* have established a good release workflow (release engineering)
* is committed to handling breaking changes
* have a clear separation of changes between configs / CLI / internal architecture

**Reply by hyf0** [reply]
I may cause some misunderstanding due to the title. This discussion isn't about when and how to release 1.0.0.

It's about how we iterate versions before 1.0.0.

**Reply by hyf0** [reply]
I change the title from `Release plan before 1.0.0` to `Release plan for 0.X.Y`.

**Comment by hyf0** [comment, marked as answer]
Sync up:

To be specific, the release plan is only for 0.X.Y version which is before 1.0.0. We would strictly follow the [semver](https://semver.org/) conversion and review the release plan after Rspack reaches 1.0.0.

### Must not introduce any breaking changes in patch versions.

### Allow breaking changes in minor versions.

### The minimum rules for deciding what changes are breaking:

- The change of Public API is not backward-compatible
- Behaviors changes are reviewed case by case, but there are no commitments for it.

### Minor releases are released per month basically
